src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefLocalTests.cs

```diff
@@ -1697,5 +1698,31 @@ static void M()
             Assert.Null(model.GetSymbolInfo(refInt).Symbol);
             Assert.Null(model.GetTypeInfo(refInt).Type);
         }
+
+        [WorkItem(17395, "https://github.com/dotnet/roslyn/issues/17453")]
+        [Fact]
+        public void Regression17395()
+        {
+            var source = @"
+using System;
+
+public class C 
+{            
+    public void F()
+    {
+        ref int[] a = ref {1,2,3};
```

Comment:
There are many scenarios with valid or invalid RHS or a ref assignment. We were specifically missing array initializer since array initializers are bound in a special way. Other initializers just do "BindValue(. . , valueKind,..)" 

---
In reply to: [103767488](https://github.com/dotnet/roslyn/pull/17477#discussion_r103767488) [](ancestors = 103767488)